src/acl/agent.ts

```diff
@@ -37,59 +37,61 @@ import {
  *
  * Also, please note that this function is still experimental: its API can change in non-major releases.
  *
- * @param dataset The LitDataset to which the given Agent may have been granted access.
+ * @param resourceInfo Information about the Resource to which the given Agent may have been granted access.
  * @param agent WebID of the Agent for which to retrieve what access it has to the Resource.
  * @returns Which Access Modes have been granted to the Agent specifically for the given LitDataset, or `null` if it could not be determined (e.g. because the current user does not have Control Access to a given Resource or its Container).
  */
 export function unstable_getAgentAccessModesOne(
-  dataset: LitDataset & WithResourceInfo & unstable_Acl,
+  resourceInfo: unstable_WithAcl,
   agent: WebId
 ): unstable_AccessModes | null {
-  if (unstable_hasResourceAcl(dataset)) {
-    const resourceAcl = unstable_getResourceAcl(dataset);
-    return unstable_getAgentResourceAccessModesOne(resourceAcl, agent);
+  if (unstable_hasResourceAcl(resourceInfo)) {
+    return unstable_getAgentResourceAccessModesOne(
+      resourceInfo.acl.resourceAcl,
+      agent
+    );
   }
-  if (unstable_hasFallbackAcl(dataset)) {
-    const fallbackAcl = unstable_getFallbackAcl(dataset);
-    return unstable_getAgentDefaultAccessModesOne(fallbackAcl, agent);
+  if (unstable_hasFallbackAcl(resourceInfo)) {
+    return unstable_getAgentDefaultAccessModesOne(
+      resourceInfo.acl.fallbackAcl,
+      agent
+    );
   }
   return null;
 }
 
 /**
- * Find out what Access Modes have been granted to specific Agents for a given LitDataset.
+ * Find out what Access Modes have been granted to specific Agents for a given Resource.
  *
  * Keep in mind that this function will not tell you what access arbitrary Agents might have through other ACL rules, e.g. public or group-specific permissions.
  *
  * Also, please note that this function is still experimental: its API can change in non-major releases.
  *
- * @param dataset The LitDataset to which Agents may have been granted access.
+ * @param resourceInfo Information about the Resource to which Agents may have been granted access.
  * @returns Which Access Modes have been granted to which Agents specifically for the given LitDataset, or `null` if it could not be determined (e.g. because the current user does not have Control Access to a given Resource or its Container).
  */
 export function unstable_getAgentAccessModesAll(
-  dataset: LitDataset & WithResourceInfo & unstable_Acl
+  resourceInfo: unstable_WithAcl
 ): unstable_AgentAccess | null {
-  if (unstable_hasResourceAcl(dataset)) {
-    const resourceAcl = unstable_getResourceAcl(dataset);
-    return unstable_getAgentResourceAccessModesAll(resourceAcl);
+  if (unstable_hasResourceAcl(resourceInfo)) {
+    return unstable_getAgentResourceAccessModesAll(
+      resourceInfo.acl.resourceAcl
```

Comment:
Hmm, was there a specific reason to remove use of `getResourceAcl` here and `getFallbackAcl` on line 94? Right now they're simple wrappers to access a property, but if they ever do change to be part of the same Dataset, using those functions gives us a single place to update.

Reply:
No specific reason, I assume it was a overzealous cleanup when changing the return type. Fixed in 74af0a1a3c930ddb98887f848aa8648689581a0e